File: jetpack_java/src/main/java/com/kunminx/jetpack_java/sample_04_databinding/ui/adapter/DataBindingMomentAdapter.java


package com.kunminx.jetpack_java.sample_04_databinding.ui.adapter;

import android.content.Context;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.DiffUtil;
import androidx.recyclerview.widget.RecyclerView;

import com.kunminx.jetpack_java.R;
import com.kunminx.jetpack_java.common_data.bean.Moment;
import com.kunminx.jetpack_java.common_ui.adapter.DiffUtilCallbacks;
import com.kunminx.jetpack_java.databinding.AdapterMomentDatabindingBinding;
import com.kunminx.architecture.ui.adapter.SimpleBindingAdapter;

/**
 * Create by KunMinX at 2020/5/31
 */
public class DataBindingMomentAdapter extends SimpleBindingAdapter<Moment, AdapterMomentDatabindingBinding> {

    public DataBindingMomentAdapter(Context context) {
        super(context, R.layout.adapter_moment_databinding, new DiffUtilCallbacks().getMomentItemCallback());
    }

    @Override
    protected void onBindItem(AdapterMomentDatabindingBinding binding, Moment item, RecyclerView.ViewHolder holder) {
        binding.setMoment(item);
    }
}

File: jetpack_kotlin/src/main/java/com/flywith24/jetpack_kotlin/sample_04_databinding/ui/adapter/DataBindingMomentAdapter.kt
package com.flywith24.jetpack_kotlin.sample_04_databinding.ui.adapter

import android.content.Context
import androidx.recyclerview.widget.RecyclerView
import com.flywith24.jetpack_kotlin.R
import com.flywith24.jetpack_kotlin.common_data.bean.Moment
import com.flywith24.jetpack_kotlin.common_ui.adapter.MomentDiffCallback
import com.flywith24.jetpack_kotlin.databinding.KotlinAdapterMomentDatabindingBinding
import com.kunminx.architecture.ui.adapter.SimpleBindingAdapter

/**
 * @author Flywith24
 * @date   2020/5/31
 * time   22:56
 * description
 */
class DataBindingMomentAdapter(context: Context) : SimpleBindingAdapter<Moment, KotlinAdapterMomentDatabindingBinding>(context, R.layout.kotlin_adapter_moment_databinding, MomentDiffCallback) {
    override fun onBindItem(binding: KotlinAdapterMomentDatabindingBinding?, item: Moment?, holder: RecyclerView.ViewHolder?) {
        binding?.moment = item
    }
}